rascal-vscode-extension/src/lsp/ParameterizedLanguageServer.ts

```diff
@@ -144,19 +144,36 @@ export class ParameterizedLanguageServer implements vscode.Disposable {
 
 }
 
-
+/**
+ * Define a DSL that the DSL multiplexer has to load
+ */
 export interface LanguageParameter {
-    pathConfig: string 		// rascal pathConfig constructor as a string
-    name: string; 			// name of the language
-    extension:string; 		// extension for files in this language
-    mainModule: string; 	// main module to locate mainFunction in
-    mainFunction: string; 	// main function which contributes the language implementation
-    precompiledParser: ParserSpecification | undefined; // optionally configure a precompiled parser function
+    /** rascal pathConfig constructor as a string */
+    pathConfig: string
+    /** name of the language */
+    name: string;
+    /** extension for files in this language */
+    extension:string;
+    /** main module to locate mainFunction in */
+    mainModule: string;
+    /** main function which contributes the language implementation */
+    mainFunction: string;
+    /** optionally configure a precompiled parser function */
+    precompiledParser: ParserSpecification | undefined;
 }
 
+/**
+ * Define precompiled parser to load before the modules have loaded.
+ * This helps reduce the time an IDE is shown without syntax highlighting
+ */
 export interface ParserSpecification {
-    parserFile: string; // which file contains the grammars
-    startTerminal: string; // which terminal to extract from the list of parsers in that file
+    /** absolute path to the file containing the precompiled parsers */
+    parserFile: string;
+    /** terminal to use from the defined parsers */
```

Comment:
that should be `non-terminal`

Reply:
🤦🏼 I always get them mixed up. or should we call them `Symbol`?

Reply:
No Paul points out, rightly so, that `Symbol` is completely generic in computer science so does not mean much at all. `Non-terminal` is the name of the concept in all books about context-free grammars and parsing. It signifies the parser has yet to expand more structure, recursively, to be able to recognize a sentence. As opposed to a `terminal`, which is an end-point of grammar rule expansion. You see why they call them "railroad diagrams" too. It's borrowed from the railroad jargon.